EEJT-000A/main/java/task2/SearchingController.java

```diff
@@ -0,0 +1,46 @@
+package task2;
+
+import java.util.ArrayList;
+import java.util.LinkedList;
+import java.util.List;
+
+import static java.util.Objects.isNull;
+
+public class SearchingController {
+    private Parameters parameters;
+    private SearchByParam searchByParam;
+
+    public SearchingController(Parameters parameters) {
+        this.parameters = parameters;
+    }
+
+    public List<String> doChain(List<String> list) {
+        if (!isNull(parameters)) {
+            if (isNull(list)) {
+                list = new ArrayList<>();
+            }
+            if ((parameters.getSizeMore()) != null
+                    && (parameters.getSizeLess()) != null) {
+                SearchByParam newSearchByParam = null;
+                searchByParam = new SearchBySizeChange(newSearchByParam);
+            }
+            if ((parameters.getDateMore()) != null
+                    && (parameters.getDateLess()) != null) {
+                SearchByParam newSearchByParam = searchByParam;
+                searchByParam = new SearchByDateChange(newSearchByParam);
```

Comment:
I believe you can replace two lines with one line assignement, like this(need to double check):
```suggestion
                searchByParam = new SearchByDateChange(searchByParam);
```